astropy/tests/test_logger.py

```diff
@@ -362,11 +362,11 @@ def test_log_to_list_origin2():
 
 
 @pytest.mark.parametrize(('level'), [None, 'DEBUG', 'INFO', 'WARN', 'ERROR'])
-def test_log_to_file(tmpdir, level):
+def test_log_to_file(tmp_path, level):
 
-    local_path = tmpdir.join('test.log')
+    local_path = tmp_path / 'test.log'
     log_file = local_path.open('wb')
-    log_path = str(local_path.realpath())
+    log_path = str(local_path)
```

Comment:
This might be closer match. Maybe makes no difference but doesn't hurt.
```suggestion
    log_path = str(local_path.resolve())
```